double_test.go

```diff
@@ -42,6 +42,28 @@ func TestEncDouble(t *testing.T) {
 	t.Logf("decode(%v) = %v, %v\n", v, res, err)
 }
 
+func TestIssue181(t *testing.T) {
+	var (
+		v   float32
+		err error
+		e   *Encoder
+		d   *Decoder
+		res interface{}
+	)
+
+	e = NewEncoder()
+	v = 99.8
+	e.Encode(v)
+	if len(e.Buffer()) == 0 {
+		t.Fail()
+	}
+
+	// res would be '99.800003' without patches in PR #196
+	d = NewDecoder(e.Buffer())
+	res, err = d.Decode()
+	t.Logf("decode(%v) = %v, %v\n", v, res, err)
+}
+
```

Comment:
I followed the previous test case `func TestEncDouble` to add the test case for Issue181. But I am puzzled by that only one judgement on `Encode()` and no judgement on if the result is right.

Reply:
yes, u r right, it should add a judgement for the result too, pls add for it